src/reactive/prometheus-apt-exporter.py

```diff
@@ -39,20 +35,24 @@ def install_packages():
     hookenv.status_set("maintenance", "Installing software")
     config = hookenv.config()
     channel = config.get("snap_channel")
+    port_number = config.get("port")
     snap.install(SNAP_NAME, channel=channel, force_dangerous=False)
-    subprocess.check_call(["snap", "connect", "prometheus-apt-exporter:apt-exporter-files"])
+    subprocess.check_call(
+        ["snap", "connect", "prometheus-apt-exporter:host-apt-contents"]
+    )
+    set_http_port()
     hookenv.status_set("active", "Exporter installed and connected")
-    hookenv.open_port(PORT_NUMBER)
-    set_state("apt-exporter.installed")
+    hookenv.open_port(port_number)
```

Comment:
I think it's better to call this in `set_http_port` function. Along with `hookenv.close_port()` on the old port value. Otherwise the open ports will just keep appending.